alarm_req_listener.cpp

```diff
@@ -123,11 +126,25 @@ bool AlarmReqListener::zmq_init_sck()
     return false;
   }
 
-  std::stringstream ss;
-  ss << "tcp://127.0.0.1:" << ZMQ_PORT; 
+  std::string sf = std::string("/var/run/clearwater/alarms");
+  std::string ss = std::string("ipc://" + sf);
+  snmp_log(LOG_INFO, "AlarmReqListener: ss='%s'", ss.c_str());
 
   int rc;
-  while (((rc = zmq_bind(_sck, ss.str().c_str())) == -1) && (errno == EINTR))
+  rc=remove(sf.c_str());
+  if (rc == -1)
+  {
+    if (errno != ENOENT)
+    {
+      snmp_log(LOG_ERR, "remove(%s) failed: %s - killing myself", sf.c_str(), strerror(errno));
+      kill(getpid(), SIGKILL);
+    }
+    else
+    {
+      snmp_log(LOG_ERR, "remove(%s) failed: %s", sf.c_str(), strerror(errno));
+    }
+  }
+  while (((rc = zmq_bind(_sck, ss.c_str())) == -1) && (errno == EINTR))
```

Comment:
My understanding is that `zmq_bind` will unlink the socket for you, so the entire preceding code block is unnecessarily.